src/error.ts

```diff
@@ -0,0 +1,47 @@
+interface FirebaseErrorOptions {
+  children?: unknown[];
+  context?: unknown;
+  exit?: number;
+  original?: Error;
+  status?: number;
+}
+
+const DEFAULT_CHILDREN: NonNullable<FirebaseErrorOptions["children"]> = [];
+const DEFAULT_EXIT: NonNullable<FirebaseErrorOptions["exit"]> = 1;
+const DEFAULT_STATUS: NonNullable<FirebaseErrorOptions["status"]> = 500;
+
+export class FirebaseError extends Error implements FirebaseErrorOptions {
```

Comment:
I think a small test file to simply test the constructor with and without `options` (and seeing the defaults/overrides) would be great.

Reply:
5b25ee0d6e89a7e49467ff79eba747861e622389